test/metabase/upload_test.clj

```diff
@@ -402,20 +418,12 @@
       (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
         (with-mysql-local-infile-on-and-off
          (with-upload-table!
-           [table (let [table-name (mt/random-name)]
-                    (@#'upload/create-from-csv!
-                     driver/*driver*
-                     (mt/id)
-                     table-name
-                     (csv-file-with lines))
-                    (sync-upload-test-table! :database (mt/db) :table-name table-name))]
+           [table (create-from-csv-and-sync-with-defaults!
+                   :file (csv-file-with lines)
+                   :auxiliary-sync-steps :synchronous)]
            (testing "Table and Fields exist after sync"
              (is (=? {:name          #"(?i)_mb_row_id"
-                      :semantic_type (if (= driver/*driver* :redshift)
-                                       ;; TODO: there is a bug in the redshift driver where the semantic_type is not set
-                                       ;; to type/PK even if the column is a PK
-                                       :type/Category
-                                       :type/PK)
+                      :semantic_type :type/PK
```

Comment:
🎉

Reply:
Yeah, I fixed this on a sync PR by accident :D